### PR TITLE
i hope i did it

### PR DESCRIPTION
**Pull request by puf17640**
way too drunk, please tell me its alright

### CODE REVIEW

**Comment by puf17640** [comment]
lma othis aint even it

**Comment by puf17640** [comment]
oh yeah now we talkin

**Comment by puf17640** [comment]
forgot how to request review, please do so @ebwinters

**Comment by puf17640** [comment]
this a good pr forreal tho

**Comment by ebwinters** [comment]
ok dude im looking jeez

File: routes/api_worldometers.js [hunk 1]
```diff
@@ -12,9 +12,9 @@ const { redis, keys } = require('./instances');
 const getAllData = async (key) => {
 	const countries = JSON.parse(await redis.get(key));
 	const worldData = countries.find(country => country.country.toLowerCase() === 'world');
+	worldData.affectedCountries = countries.filter(country => country.country.toLowerCase() !== 'world').length;
 	delete worldData.country;
 	delete worldData.countryInfo;
-	worldData.affectedCountries = countries.splice(1).length;
 	return worldData;
```
Comment (ebwinters) [inline routes/api_worldometers.js:18]: object de-structuring preferred:
```js
const { country, countryInfo, ...cleanedWorldData } = worldData;
return cleanedWorldData
```

File: utils/logger.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
 module.exports = {
-	httpErrorLogger
+	err: (message = 'Unknown Error', err) => console.error({
+		message: `[${new Date().toISOString()}]: ${message}`,
+		error: err.message,
+		stack: err.stack
+	}),
+	info: (message) => console.info(`[${new Date().toISOString()}]: ${message}`),
+	warn: (message) => console.warn(`[${new Date().toISOString()}]: ${message}`)
 };
```
Comment (ebwinters) [inline utils/logger.js:10]: Better practice to move all of these to functions. So like 
```js
const err = (...) => {};
const info = (...) => {};
const warn = (...) => {}
module.exports = {
  err,
  info,
  warn
};
```

File: scrapers/getWorldometers.js [hunk 3]
```diff
@@ -3,13 +3,14 @@ const cheerio = require('cheerio');
...
 const logger = require('../utils/logger');
 
-/**
- * Gets country data from table cell on worldometers website
- * @param 	{Object} 	cell 	Country table cell from worldometers website
- * @returns {string} 			Country name
- */
-const getCountryData = (cell) => (cell.children[0].data || cell.children[0].children[0].data || cell.children[0].children[0].children[0].data
-	|| cell.children[0].children[0].children[0].children[0].data || (cell.children[0].next.children[0] && cell.children[0].next.children[0].data) || '').trim();
+const columnMapper = ['cases', 'todayCases', 'deaths', 'todayDeaths', 'recovered', 'active', 'critical', 'casesPerOneMillion', 'deathsPerOneMillion', 'tests', 'testsPerOneMillion'];
```
Comment (ebwinters) [inline scrapers/getWorldometers.js:6]: just call this `columns`, `columnMapper` sounds like a function name

File: scrapers/getWorldometers.js [hunk 4]
```diff
@@ -3,13 +3,14 @@ const cheerio = require('cheerio');
...
+
+const continentMapping = (element) => {
+	element.continent = element.country;
+	delete element.country;
+	delete element.countryInfo;
+	return element;
```
Comment (ebwinters) [inline scrapers/getWorldometers.js:12]: change for obj de-structuring
```js
element.continent = element.country;
const { country, countryInfo, ...cleanedElement } = element;
return element;
```

File: scrapers/getWorldometers.js [hunk 5]
```diff
@@ -3,13 +3,14 @@ const cheerio = require('cheerio');
...
-/**
- * Gets country data from table cell on worldometers website
- * @param 	{Object} 	cell 	Country table cell from worldometers website
- * @returns {string} 			Country name
- */
-const getCountryData = (cell) => (cell.children[0].data || cell.children[0].children[0].data || cell.children[0].children[0].children[0].data
-	|| cell.children[0].children[0].children[0].children[0].data || (cell.children[0].next.children[0] && cell.children[0].next.children[0].data) || '').trim();
+const columnMapper = ['cases', 'todayCases', 'deaths', 'todayDeaths', 'recovered', 'active', 'critical', 'casesPerOneMillion', 'deathsPerOneMillion', 'tests', 'testsPerOneMillion'];
+
+const continentMapping = (element) => {
```
Comment (ebwinters) [inline scrapers/getWorldometers.js:8]: Add JSDOC

File: scrapers/getWorldometers.js [hunk 6]
```diff
@@ -18,103 +19,36 @@ const getCountryData = (cell) => (cell.children[0].data || cell.children[0].chil
 */
 const getOrderByCountryName = (data) => data.sort((a, b) => a.country < b.country ? -1 : 1);
 
-/**
- * Gets integer parsed stat from table cell
- * @param 	{Object} 	cell 	Table cell from worldometers website
- * @returns {number} 			Number from cell for statistic
- */
-const getCellData = (cell) => parseInt((cell.children.length !== 0 ? cell.children[0].data : '').trim().replace(/,/g, '') || '0', 10);
+const mapRows = (_, row) => {
```
Comment (ebwinters) [inline scrapers/getWorldometers.js:22]: Add JSDOC

File: scrapers/getWorldometers.js [hunk 7]
```diff
@@ -18,103 +19,36 @@ const getCountryData = (cell) => (cell.children[0].data || cell.children[0].chil
...
+	});
+	return country;
+};
+
 /**
  * Fills an array full of table data parsed from worldometers
  * @param 	{Object} 	html 		Cheerio HTML object from worldometers site
- * @param 	{boolean} 	yesterday 	Default false, tells the function which URL to use data from
- * @returns {array} 				Array of objects containing table data from worldometers
+ * @param 	{boolean} 	idExtension 	the extension to append to the ID that is used to get the tables (either 'today' or 'yesterday')
```
Comment (ebwinters) [inline scrapers/getWorldometers.js:41]: this is a string not a boolean

File: scrapers/getWorldometers.js [hunk 8]
```diff
@@ -18,103 +19,36 @@ const getCountryData = (cell) => (cell.children[0].data || cell.children[0].chil
...
-					country.testsPerOneMillion = getCellData(cell) || 0;
-					break;
-			}
-		}
-		return country;
-	}).get();
-	const world = countriesData.find(country => country.country.toLowerCase() === 'world');
+function fillResult(html, idExtension) {
+	const countriesTable = html(`table#main_table_countries_${idExtension}`);
+	const countriesData = countriesTable.children('tbody:first-of-type').children('tr:not(.row_continent)').map(mapRows).get();
```
Comment (ebwinters) [inline scrapers/getWorldometers.js:46]: change variable name to `countries` so we can return {.., countries, ..}

File: scrapers/getWorldometers.js [hunk 9]
```diff
@@ -18,103 +19,36 @@ const getCountryData = (cell) => (cell.children[0].data || cell.children[0].chil
...
+	const continentData = countriesTable.children('tbody:first-of-type').children('tr.row_continent').map(mapRows).get().map(continentMapping).filter(data => !!data.continent);
```
Comment (ebwinters) [inline scrapers/getWorldometers.js:47]: change name to `continents` so we can return {.., continents, ..}

File: scrapers/getWorldometers.js [hunk 10]
```diff
@@ -18,103 +19,36 @@ const getCountryData = (cell) => (cell.children[0].data || cell.children[0].chil
...
+	const world = countriesData.shift();
 	world.tests = countriesData.map(country => country.tests).splice(1).reduce((sum, test) => sum + test);
 	world.testsPerOneMillion = parseFloat(((1e6 / (1e6 / (world.casesPerOneMillion / world.cases))) * world.tests).toFixed(1));
-	return countriesData;
+	return { world, countries: countriesData, continents: continentData };
 }
 
 /**
```
Comment (ebwinters) [inline scrapers/getWorldometers.js:54]: Update JSDOC description to include continents

File: utils/string_utils.js [hunk 11]
```diff
@@ -1,21 +1,5 @@
...
-
-const splitQuery = (query) => query.indexOf('|') === -1 ? query.split(',') : query.split('|');
-
 module.exports = {
-	wordsStandardize,
-	wordToBoolean,
-	splitQuery
+	wordsStandardize: (word) => word ? word.trim().toLowerCase().normalize('NFD').replace(/[\u0300-\u036f]/g, '') : 'null',
+	wordToBoolean: (word) => word === '' || (word ? word.toLowerCase() === 'true' || word === '1' : false),
+	splitQuery: (query) => query.indexOf('|') === -1 ? query.split(',') : query.split('|')
```
Comment (ebwinters) [inline utils/string_utils.js:4]: It's good to shorten these functions, but they should be declared as const and then exported. It's better practice.
```js
wordsStandardize = (word) => { ... };
wordToBoolean = (word) => { ... };
splitQuery = (word) => { ... };
module.exports = {
  wordsStandardize,
  wordToBoolean,
  splitQuery
}

File: utils/country_utils.js [hunk 12]
```diff
@@ -6,122 +6,67 @@ const countryData = require('./countries');
...
-		// number, must be country ID
-		return ctry.countryInfo._id === Number(countryNameParam);
-	});
-	return foundCountry;
+const getWorldometersData = (data, nameParam, strictMatching, continentMode) => {
+	const selector = continentMode ? 'continent' : 'country';
+	const isText = isNaN(nameParam);
+	const countryInfo = isText ? getCountryData(nameParam) : {};
+	const standardizedName = stringUtils.wordsStandardize(countryInfo.country ? countryInfo.country : nameParam);
+	return data.find((ctry) => !isText ? ctry.countryInfo && ctry.countryInfo._id === Number(nameParam) : strictMatching ? stringUtils.wordsStandardize(ctry[selector]) === standardizedName : ((ctry.countryInfo || {}).iso3 || '').toLowerCase() === nameParam.toLowerCase()
```
Comment (ebwinters) [inline utils/country_utils.js:53]: just split this to more lines and remove `/* eslint-disable max-len */` at the top of the file. It's better not to disable linting when we can

File: utils/country_utils.js [hunk 13]
```diff
@@ -6,122 +6,67 @@ const countryData = require('./countries');
...
-		}
-	}
-	return false;
-};
+/**
+ * Check for country exception when searching countries
+ * @param {string} countryname name of the country to be searched
+ * @returns {boolean}
+ */
+const isCountryException = (countryname) => !!['UK', 'UAE', 'DR'].find(exception => stringUtils.wordsStandardize(countryname) === stringUtils.wordsStandardize(exception));
```
Comment (ebwinters) [inline utils/country_utils.js:64]: put `['UK', 'UAE', 'DR']` into a variable for easier editing later

File: routes/api_worldometers.js (hunk 1 above)
Comment (puf17640) [reply routes/api_worldometers.js:18]: omg i love this

File: utils/logger.js (hunk 2 above)
Comment (puf17640) [reply utils/logger.js:10]: very clever

File: scrapers/getWorldometers.js (hunk 3 above)
Comment (puf17640) [reply scrapers/getWorldometers.js:6]: exactly

File: scrapers/getWorldometers.js (hunk 4 above)
Comment (puf17640) [reply scrapers/getWorldometers.js:12]: will do

**Comment by puf17640** [comment]
will work on this tomorrow, thanks for the feedback

**Comment by buster95** [comment]
sos un crack

**Pull request merged**